src/login/logind-dbus.c

```diff
@@ -920,7 +916,11 @@ static int method_create_session(sd_bus_message *message, void *userdata, sd_bus
                         goto fail;
         }
 
-        r = session_start(session);
+        r = sd_bus_message_enter_container(message, 'a', "(sv)");
+        if (r < 0)
+                return r;
+
+        r = session_start(session, message);
         if (r < 0)
                 goto fail;
```

Comment:
it would be nice if you'd leave the container again after this, just as an extra safety check that everything is in order.